### ISSUE TITLE
WebSocketServerFactory NPE in toString()

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#1603]
The WebSocketServerFactory.toString() can throw a NPE during startup.

**Comment by sbordet** [comment]
```
java.lang.NullPointerException
	at org.eclipse.jetty.websocket.server.WebSocketServerFactory.toString(WebSocketServerFactory.java:641)
	at org.slf4j.helpers.MessageFormatter.safeObjectAppend(MessageFormatter.java:299)
	at org.slf4j.helpers.MessageFormatter.deeplyAppendParameter(MessageFormatter.java:271)
	at org.slf4j.helpers.MessageFormatter.arrayFormat(MessageFormatter.java:233)
	at org.slf4j.helpers.MessageFormatter.arrayFormat(MessageFormatter.java:173)
	at org.eclipse.jetty.util.log.JettyAwareLogger.log(JettyAwareLogger.java:619)
	at org.eclipse.jetty.util.log.JettyAwareLogger.debug(JettyAwareLogger.java:206)
	at org.eclipse.jetty.util.log.Slf4jLog.debug(Slf4jLog.java:89)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.addBean(ContainerLifeCycle.java:322)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.addBean(ContainerLifeCycle.java:226)
	at org.eclipse.jetty.websocket.server.WebSocketServerFactory.<init>(WebSocketServerFactory.java:149)
	at org.eclipse.jetty.websocket.server.WebSocketServerFactory.<init>(WebSocketServerFactory.java:126)
	at org.eclipse.jetty.websocket.server.WebSocketServerFactory.<init>(WebSocketServerFactory.java:105)
	at org.eclipse.jetty.websocket.server.NativeWebSocketConfiguration.<init>(NativeWebSocketConfiguration.java:51)
	at org.eclipse.jetty.websocket.server.NativeWebSocketServletContainerInitializer.getDefaultFrom(NativeWebSocketServletContainerInitializer.java:36)
	at org.eclipse.jetty.websocket.jsr356.server.deploy.WebSocketServerContainerInitializer.configureContext(WebSocketServerContainerInitializer.java:139)
```

**Comment by joakime** [comment]
Fixed in commit c1143fc

**Comment by bmahaj2** [comment]
On which version of Jetty is this bug fixed ? I'm using spring version 1.5.7 which uses jetty 9.4.6.v20170531 and I see this bug. How do I make it work with spring 1.5.7 ?

 org.springframework.boot:spring-boot-starter-jetty:jar:1.5.7.RELEASE:compile
[INFO] |  +- org.eclipse.jetty:jetty-servlets:jar:9.4.6.v20170531:compile

**Comment by joakime** [comment]
Looking at the commit associated with this issue ...

Commit ab98caafbb4f8874dc2e2c84ab440983fa0d70e1

We can see that it first shows up in jetty-9.4.7.v20170914
